lib/resolver.js

```diff
@@ -84,7 +84,7 @@ class SchemaResolver {
     resolveReference(value) {
         let refschema;
 
-        const id = value.substr(0, value.indexOf('#') + 1);
```

Comment:
The `id` is identifier of the reference. see https://json-spec.readthedocs.io/reference.html#json-reference for example. This code change would break that.

Reply:
Perhaps I'm not using it correctly, but when I just run that part of the function given the pattern suggested in the docs (`#sub/d`), the id resolves to `#`.

Then later, when the following tries to use the id:
```js
if (id && this.subSchemas) {
    refschema = this.subSchemas[id] || this.subSchemas[id.substr(0, id.length - 1)]
  }
  if (!refschema) {
    refschema = this.root
  }
```
`this.subSchemas[id]` will never find the correct reference.

Am I missing something?

Reply:
`Id` will always resolve to `#`. That can't be the desired result, can it?